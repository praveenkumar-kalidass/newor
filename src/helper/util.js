import CONSTANT from 'constant';

export const formatCurrency = (value) => {
  const formatter = new Intl.NumberFormat('hi-IN', { style: 'currency', currency: 'INR' });
  return formatter.format(value);
};

export const unformatCurrency = (value) => {
  const withoutSymbol = value.replace(CONSTANT.APP_LITERAL.RUPEE_SYMBOL, '');
  const currency = withoutSymbol.replace(/,/g, '');
  return parseFloat(currency);
};

export const formatUrlParams = (data) => Object.keys(data).map((key) => (
  `${encodeURIComponent(key)}=${encodeURIComponent(data[key])}`
)).join('&');
