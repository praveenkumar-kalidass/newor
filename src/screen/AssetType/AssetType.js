import React, { useMemo } from 'react';
import { TouchableOpacity } from 'react-native';
import { useNavigation } from '@react-navigation/native';
import {
  ScrollView, VStack, Text, Divider, Box, Badge, HStack,
} from 'native-base';
import FontAwesome from 'react-native-vector-icons/FontAwesome';

import useTranslation from 'translation/useTranslation';
import useTheme from 'theme/useTheme';
import CONSTANT from 'constant';
import COLOR from 'constant/color';
import ROUTE from 'constant/route';
import { AssetCard } from './AssetType.style';

const AssetType = () => {
  const navigation = useNavigation();
  const { translate } = useTranslation();
  const theme = useTheme();
  const assetTypes = useMemo(() => [
    {
      label: translate('DEPOSIT'),
      list: Object.values(CONSTANT.DEPOSIT_TYPE).map((value, index) => ({
        label: translate(value),
        color: COLOR.RANDOM[index],
      })),
      onPress: () => navigation.navigate(ROUTE.ADD_DEPOSIT),
    },
  ], []);

  return (
    <ScrollView>
      <VStack p={5}>
        <For each="assetType" index="index" of={assetTypes}>
          <TouchableOpacity
            key={`asset-type-${index}`}
            testID={`asset-type-${assetType.label}`}
            onPress={assetType.onPress}
          >
            <AssetCard>
              <VStack flex>
                <Text fontSize={16} bold>{assetType.label}</Text>
                <Divider my={3} />
                <HStack flexWrap>
                  <For each="item" index="itemIndex" of={assetType.list}>
                    <Badge
                      key={`asset-type-${index}-item-${itemIndex}`}
                      bg={item.color}
                      m={1}
                    >
                      {item.label}
                    </Badge>
                  </For>
                </HStack>
              </VStack>
              <Box pl={5} py={1}>
                <FontAwesome color={theme.color.BACKGROUND_0} size={20} name="chevron-right" />
              </Box>
            </AssetCard>
          </TouchableOpacity>
        </For>
      </VStack>
    </ScrollView>
  );
};

export default AssetType;
