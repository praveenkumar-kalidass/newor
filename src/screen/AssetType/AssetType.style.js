import styled from 'styled-components/native';
import { HStack } from 'native-base';

// eslint-disable-next-line import/prefer-default-export
export const AssetCard = styled(HStack).attrs(({ theme }) => ({
  bg: theme.color.MODAL,
  p: 5,
  rounded: 10,
}))``;
