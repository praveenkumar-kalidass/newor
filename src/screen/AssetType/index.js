import AssetType from './AssetType';

export default AssetType;
