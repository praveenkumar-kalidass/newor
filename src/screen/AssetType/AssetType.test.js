import React from 'react';

import { render } from 'test/util';
import AssetType from './AssetType';

describe('AssetType', () => {
  it('should match snapshot', () => {
    const container = render(<AssetType />);

    expect(container).toMatchSnapshot();
  });
});
