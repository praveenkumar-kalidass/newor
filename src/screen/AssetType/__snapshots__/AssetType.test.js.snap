// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssetType should match snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Array [
          Object {
            "borderWidth": 0,
            "left": 0,
            "position": "absolute",
            "top": 0,
          },
          Object {
            "height": 1334,
            "width": 750,
          },
        ]
      }
    >
      <RCTScrollView
        contentContainerStyle={
          Array [
            Object {},
            Object {
              "dataSet": Object {},
            },
          ]
        }
        dataSet={Object {}}
        style={Object {}}
      >
        <View>
          <View
            dataSet={Object {}}
            style={
              Object {
                "paddingBottom": 20,
                "paddingLeft": 20,
                "paddingRight": 20,
                "paddingTop": 20,
              }
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID="asset-type-DEPOSIT"
            >
              <View
                dataSet={Object {}}
                style={
                  Array [
                    Object {
                      "backgroundColor": "#eeeeee",
                      "borderRadius": 10,
                      "flexDirection": "row",
                      "paddingBottom": 20,
                      "paddingLeft": 20,
                      "paddingRight": 20,
                      "paddingTop": 20,
                    },
                    Array [
                      Object {},
                    ],
                  ]
                }
              >
                <View
                  dataSet={Object {}}
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                >
                  <Text
                    dataSet={Object {}}
                    style={
                      Object {
                        "backgroundColor": undefined,
                        "color": "#171717",
                        "fontFamily": undefined,
                        "fontSize": 16,
                        "fontStyle": "normal",
                        "fontWeight": "700",
                        "letterSpacing": 0,
                        "lineHeight": 24,
                        "textDecorationLine": undefined,
                      }
                    }
                  >
                    DEPOSIT
                  </Text>
                  <View
                    dataSet={Object {}}
                    style={Object {}}
                  />
                  <View
                    aria-orientation="horizontal"
                    dataSet={Object {}}
                    style={
                      Object {
                        "backgroundColor": "#d4d4d4",
                        "height": 1,
                        "marginBottom": 12,
                        "marginTop": 12,
                        "width": "100%",
                      }
                    }
                    thickness="1"
                  />
                  <View
                    dataSet={Object {}}
                    style={Object {}}
                  />
                  <View
                    dataSet={Object {}}
                    style={
                      Object {
                        "flexDirection": "row",
                        "flexWrap": "wrap",
                      }
                    }
                  >
                    <View
                      colorScheme="muted"
                      dataSet={Object {}}
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#EC7063",
                          "borderColor": "transparent",
                          "borderRadius": 2,
                          "borderWidth": 1,
                          "flexDirection": "row",
                          "justifyContent": "center",
                          "marginBottom": 4,
                          "marginLeft": 4,
                          "marginRight": 4,
                          "marginTop": 4,
                          "paddingBottom": 2,
                          "paddingLeft": 8,
                          "paddingRight": 8,
                          "paddingTop": 2,
                        }
                      }
                    >
                      <View
                        dataSet={Object {}}
                        style={Object {}}
                      >
                        <Text
                          dataSet={Object {}}
                          style={
                            Object {
                              "backgroundColor": undefined,
                              "color": "#171717",
                              "fontFamily": undefined,
                              "fontSize": 12,
                              "fontStyle": "normal",
                              "fontWeight": "500",
                              "letterSpacing": 0,
                              "lineHeight": 18,
                              "textDecorationLine": undefined,
                            }
                          }
                        >
                          CONSTANT.DEPOSIT_TYPE.SAVINGS
                        </Text>
                      </View>
                    </View>
                    <View
                      dataSet={Object {}}
                      style={Object {}}
                    />
                    <View
                      colorScheme="muted"
                      dataSet={Object {}}
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#AF7AC5",
                          "borderColor": "transparent",
                          "borderRadius": 2,
                          "borderWidth": 1,
                          "flexDirection": "row",
                          "justifyContent": "center",
                          "marginBottom": 4,
                          "marginLeft": 4,
                          "marginRight": 4,
                          "marginTop": 4,
                          "paddingBottom": 2,
                          "paddingLeft": 8,
                          "paddingRight": 8,
                          "paddingTop": 2,
                        }
                      }
                    >
                      <View
                        dataSet={Object {}}
                        style={Object {}}
                      >
                        <Text
                          dataSet={Object {}}
                          style={
                            Object {
                              "backgroundColor": undefined,
                              "color": "#171717",
                              "fontFamily": undefined,
                              "fontSize": 12,
                              "fontStyle": "normal",
                              "fontWeight": "500",
                              "letterSpacing": 0,
                              "lineHeight": 18,
                              "textDecorationLine": undefined,
                            }
                          }
                        >
                          CONSTANT.DEPOSIT_TYPE.CURRENT
                        </Text>
                      </View>
                    </View>
                    <View
                      dataSet={Object {}}
                      style={Object {}}
                    />
                    <View
                      colorScheme="muted"
                      dataSet={Object {}}
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#5DADE2",
                          "borderColor": "transparent",
                          "borderRadius": 2,
                          "borderWidth": 1,
                          "flexDirection": "row",
                          "justifyContent": "center",
                          "marginBottom": 4,
                          "marginLeft": 4,
                          "marginRight": 4,
                          "marginTop": 4,
                          "paddingBottom": 2,
                          "paddingLeft": 8,
                          "paddingRight": 8,
                          "paddingTop": 2,
                        }
                      }
                    >
                      <View
                        dataSet={Object {}}
                        style={Object {}}
                      >
                        <Text
                          dataSet={Object {}}
                          style={
                            Object {
                              "backgroundColor": undefined,
                              "color": "#171717",
                              "fontFamily": undefined,
                              "fontSize": 12,
                              "fontStyle": "normal",
                              "fontWeight": "500",
                              "letterSpacing": 0,
                              "lineHeight": 18,
                              "textDecorationLine": undefined,
                            }
                          }
                        >
                          CONSTANT.DEPOSIT_TYPE.FIXED_DEPOSIT
                        </Text>
                      </View>
                    </View>
                    <View
                      dataSet={Object {}}
                      style={Object {}}
                    />
                    <View
                      colorScheme="muted"
                      dataSet={Object {}}
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#48C9B0",
                          "borderColor": "transparent",
                          "borderRadius": 2,
                          "borderWidth": 1,
                          "flexDirection": "row",
                          "justifyContent": "center",
                          "marginBottom": 4,
                          "marginLeft": 4,
                          "marginRight": 4,
                          "marginTop": 4,
                          "paddingBottom": 2,
                          "paddingLeft": 8,
                          "paddingRight": 8,
                          "paddingTop": 2,
                        }
                      }
                    >
                      <View
                        dataSet={Object {}}
                        style={Object {}}
                      >
                        <Text
                          dataSet={Object {}}
                          style={
                            Object {
                              "backgroundColor": undefined,
                              "color": "#171717",
                              "fontFamily": undefined,
                              "fontSize": 12,
                              "fontStyle": "normal",
                              "fontWeight": "500",
                              "letterSpacing": 0,
                              "lineHeight": 18,
                              "textDecorationLine": undefined,
                            }
                          }
                        >
                          CONSTANT.DEPOSIT_TYPE.RECURRING_DEPOSIT
                        </Text>
                      </View>
                    </View>
                    <View
                      dataSet={Object {}}
                      style={Object {}}
                    />
                    <View
                      colorScheme="muted"
                      dataSet={Object {}}
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#58D68D",
                          "borderColor": "transparent",
                          "borderRadius": 2,
                          "borderWidth": 1,
                          "flexDirection": "row",
                          "justifyContent": "center",
                          "marginBottom": 4,
                          "marginLeft": 4,
                          "marginRight": 4,
                          "marginTop": 4,
                          "paddingBottom": 2,
                          "paddingLeft": 8,
                          "paddingRight": 8,
                          "paddingTop": 2,
                        }
                      }
                    >
                      <View
                        dataSet={Object {}}
                        style={Object {}}
                      >
                        <Text
                          dataSet={Object {}}
                          style={
                            Object {
                              "backgroundColor": undefined,
                              "color": "#171717",
                              "fontFamily": undefined,
                              "fontSize": 12,
                              "fontStyle": "normal",
                              "fontWeight": "500",
                              "letterSpacing": 0,
                              "lineHeight": 18,
                              "textDecorationLine": undefined,
                            }
                          }
                        >
                          CONSTANT.DEPOSIT_TYPE.PUBLIC_PROVIDENT_FUND
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
                <View
                  dataSet={Object {}}
                  style={Object {}}
                />
                <View
                  dataSet={Object {}}
                  style={
                    Object {
                      "paddingBottom": 4,
                      "paddingLeft": 20,
                      "paddingTop": 4,
                    }
                  }
                >
                  <Text
                    allowFontScaling={false}
                    selectable={false}
                    style={
                      Array [
                        Object {
                          "color": "#000000",
                          "fontSize": 20,
                        },
                        undefined,
                        Object {
                          "fontFamily": "FontAwesome",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
      <View
        pointerEvents="none"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            undefined,
          ]
        }
      />
    </View>
    <View
      elevation={0}
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Array [
          Object {
            "borderWidth": 0,
            "left": -600,
            "position": "absolute",
            "top": 0,
          },
          Object {
            "height": 1334,
            "width": 600,
          },
        ]
      }
    >
      <View
        dataSet={Object {}}
        style={
          Array [
            Object {
              "backgroundColor": "#eeeeee",
              "flex": 1,
              "justifyContent": "space-between",
              "paddingBottom": 80,
              "paddingTop": 80,
            },
            Array [
              Object {},
            ],
          ]
        }
      >
        <View
          dataSet={Object {}}
          style={
            Array [
              Object {
                "borderBottomWidth": 1,
                "borderColor": "#7f7f7f",
                "paddingBottom": 12,
                "paddingLeft": 16,
                "paddingRight": 20,
                "paddingTop": 12,
              },
              Array [
                Object {},
              ],
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              dataSet={Object {}}
              style={
                Object {
                  "alignItems": "center",
                }
              }
            >
              <View
                dataSet={Object {}}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#a1a1aa",
                    "borderColor": "#27272a",
                    "borderRadius": 9999,
                    "height": 96,
                    "justifyContent": "center",
                    "position": "relative",
                    "width": 96,
                  }
                }
              >
                <Text
                  allowFontScaling={false}
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": undefined,
                        "fontSize": 40,
                      },
                      undefined,
                      Object {
                        "fontFamily": "FontAwesome",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
              </View>
              <View
                dataSet={Object {}}
                style={Object {}}
              />
              <Text
                dataSet={Object {}}
                style={
                  Object {
                    "backgroundColor": undefined,
                    "color": "#171717",
                    "fontFamily": undefined,
                    "fontSize": 18,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                    "lineHeight": 27,
                    "marginBottom": 8,
                    "marginTop": 8,
                    "textDecorationLine": undefined,
                  }
                }
              >
                undefined undefined
              </Text>
            </View>
          </View>
        </View>
        <RCTScrollView
          contentContainerStyle={
            Array [
              Object {},
              Object {
                "dataSet": Object {},
              },
            ]
          }
          data={
            Array [
              Object {
                "icon": <Icon
                  allowFontScaling={false}
                  color="#000000"
                  name="power-off"
                  size={24}
                />,
                "id": "LOGOUT",
                "label": "LOGOUT",
                "onPress": [Function],
              },
            ]
          }
          dataSet={Object {}}
          getItem={[Function]}
          getItemCount={[Function]}
          keyExtractor={[Function]}
          onContentSizeChange={[Function]}
          onLayout={[Function]}
          onMomentumScrollBegin={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEventThrottle={50}
          stickyHeaderIndices={Array []}
          style={Object {}}
          viewabilityConfigCallbackPairs={Array []}
        >
          <View>
            <View
              onLayout={[Function]}
              style={null}
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  dataSet={Object {}}
                  style={
                    Array [
                      Object {
                        "borderBottomWidth": 1,
                        "borderColor": "#7f7f7f",
                        "paddingBottom": 12,
                        "paddingLeft": 16,
                        "paddingRight": 20,
                        "paddingTop": 12,
                      },
                      Array [
                        Object {},
                      ],
                    ]
                  }
                >
                  <View
                    dataSet={Object {}}
                    style={
                      Object {
                        "flexDirection": "row",
                      }
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      selectable={false}
                      style={
                        Array [
                          Object {
                            "color": "#000000",
                            "fontSize": 24,
                          },
                          undefined,
                          Object {
                            "fontFamily": "FontAwesome",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      
                    </Text>
                    <View
                      dataSet={Object {}}
                      style={
                        Object {
                          "width": 12,
                        }
                      }
                    />
                    <View
                      dataSet={Object {}}
                      style={
                        Object {
                          "justifyContent": "center",
                        }
                      }
                    >
                      <Text
                        dataSet={Object {}}
                        style={
                          Object {
                            "backgroundColor": undefined,
                            "color": "#171717",
                            "fontFamily": undefined,
                            "fontSize": 14,
                            "fontStyle": "normal",
                            "fontWeight": "700",
                            "letterSpacing": 0,
                            "lineHeight": 21,
                            "textDecorationLine": undefined,
                          }
                        }
                      >
                        LOGOUT
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
      <View
        pointerEvents="none"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
</RNCSafeAreaProvider>
`;
